wordpress-mobile/WordPress-Android
# [Gutenberg] Gallery refactor - Update upload completion processor

### Description
This updates the Gallery block upload completion processor to be compatible with the refactored format. It works by detecting the refactored version by the presence of the `imageCount` property in the block's json attributes, and defers to the existing image block processor for the inner block contents.

To test:
Run the upload completion processor unit tests.

Manual testing:
* Create a post on the web (or a version of the mobile editor without the refactored gallery block)
* Add a gallery with at least one image
* Open this post with [app version from this PR](https://109799-9306568-gh.circle-artifacts.com/0/Artifacts/WordPress-pr-14760-build-109799.apk)
  * Expect the gallery to display as the way as the non-refactored version
* Add a new gallery with at least one image
  * Expect the new gallery to display as the refactored version
* Peform the steps from [the upload tests described here](https://github.com/wordpress-mobile/test-cases/blob/trunk/test-suites/gutenberg/gallery-upload.md) on each gallery version

## Regression Notes
1. Potential unintended areas of impact
Upload completion processing.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Existing tests are relied upon, as well as new unit tests described below.

3. What automated tests I added

Unit tests were added for individual gallery blocks and for posts, to ensure that the processors can handle mixed content, since it will be possible to encounter (and edit) both formats in the same post.

PR submission checklist:
